dali/imgcodec/image_format.cc

```diff
@@ -67,5 +67,19 @@ span<ImageFormat* const> ImageFormatRegistry::Formats() const {
   return make_cspan(format_ptrs_);
 }
 
+size_t ImageParser::ReadHeader(ImageSource *encoded, uint8_t *buffer, size_t n) const {
+  if (encoded->Kind() == InputKind::HostMemory) {
+    if (encoded->Size() < n)
+      n = encoded->Size();
+    std::copy_n(encoded->RawData<char>(), n, buffer);
```

Comment:
```suggestion
    std::memcpy(buffer, encoded->RawData<char>(), n);
```
`copy_n` copies element-by-element. Maybe the compiler will recognize that, but that's a big "if".

Reply:
Done